pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -301,13 +295,29 @@ protected void internalGrantPermissionsOnTopic(String role, Set<AuthAction> acti
             log.warn("[{}] Failed to grant permissions on topic {}: concurrent modification", clientAppId(),
                     topicUri);
             throw new RestException(Status.CONFLICT, "Concurrent modification");
-        }
-        catch (Exception e) {
+        } catch (Exception e) {
             log.error("[{}] Failed to grant permissions for topic {}", clientAppId(), topicUri, e);
             throw new RestException(e);
         }
     }
 
+    protected void internalGrantPermissionsOnTopic(String role, Set<AuthAction> actions) {
+        // This operation should be reading from zookeeper and it should be allowed without having admin privileges
+        validateAdminAccessForTenant(namespaceName.getTenant());
+        validatePoliciesReadOnlyAccess();
+
+        PartitionedTopicMetadata meta = getPartitionedTopicMetadata(topicName, true, false);
```

Comment:
Before that, we seem to add a version of judgment. I'm not sure whether the partition topic is supported in the v1 version because the v1 version of the domain path has `cluster` attribute, and the v2 version does not have this attribute, So calling function `getPartitionedTopicMetadata` directly will throw an exception. therefore, I think it may need to add a version check here. 

```
if (topicName.isV2()) {
   getPartitionedTopicMetadata(topicName, true, false);
   // Auth to parititioned topic
  ....
} else {
    // Non-partitioned topic normal authorization
}
```

REST API v1:
```
persistent://tenant/cluster-name/namespace/topic-name
```

REST API
```
persistent://tenant/namespace/topic-name
```


@sijie  I'd like to hear your opinion. I don't know whether rest API v1 supports partitioned topic or not.

Reply:
@zhaohaidao We need support rest API of v1 version for the partitioned topic.

Reply:
@tuteng As far as I understand, rest API v1 already supports partitioned topic. Can you help me check if my understanding is right.
org.apache.pulsar.broker.admin.v1.PersistentTopics has implement API about partitioned topic, like createPartitionedTopic and getPartitionedMetadata as followed
```java
    @PUT
    @Path("/{property}/{cluster}/{namespace}/{topic}/partitions")
    @ApiOperation(hidden = true, value = "Create a partitioned topic.", notes = "It needs to be called before creating a producer on a partitioned topic.")
    @ApiResponses(value = { @ApiResponse(code = 403, message = "Don't have admin permission"),
            @ApiResponse(code = 409, message = "Partitioned topic already exist") })
    public void createPartitionedTopic(@PathParam("property") String property, @PathParam("cluster") String cluster,
            @PathParam("namespace") String namespace, @PathParam("topic") @Encoded String encodedTopic,
            int numPartitions) {
        validateTopicName(property, cluster, namespace, encodedTopic);
        internalCreatePartitionedTopic(numPartitions);
    }
...
    @GET
    @Path("/{property}/{cluster}/{namespace}/{topic}/partitions")
    @ApiOperation(hidden = true, value = "Get partitioned topic metadata.")
    @ApiResponses(value = { @ApiResponse(code = 403, message = "Don't have admin permission") })
    public PartitionedTopicMetadata getPartitionedMetadata(@PathParam("property") String property,
            @PathParam("cluster") String cluster, @PathParam("namespace") String namespace,
            @PathParam("topic") @Encoded String encodedTopic,
            @QueryParam("authoritative") @DefaultValue("false") boolean authoritative,
            @QueryParam("checkAllowAutoCreation") @DefaultValue("false") boolean checkAllowAutoCreation) {
        validateTopicName(property, cluster, namespace, encodedTopic);
        return internalGetPartitionedMetadata(authoritative, checkAllowAutoCreation);
    }

```

Reply:
You are right. But one unit test never passed:

You can test it locally:

```
mvn test -Dtest=org.apache.pulsar.broker.admin.AdminTest#persistentTopics -pl pulsar-broker

```

Reply:
Thanks for reminding. I have fixed this failed case.